llvm/lib/CodeGen/TargetSchedule.cpp

```diff
@@ -168,16 +168,20 @@ static unsigned findUseIdx(const MachineInstr *MI, unsigned UseOperIdx) {
   return UseIdx;
 }
 
-// Top-level API for clients that know the operand indices.
+// Top-level API for clients that know the operand indices. This doesn't need to
+// return std::optional<unsigned>, as it always returns a valid latency.
 unsigned TargetSchedModel::computeOperandLatency(
   const MachineInstr *DefMI, unsigned DefOperIdx,
   const MachineInstr *UseMI, unsigned UseOperIdx) const {
 
+  const unsigned InstrLatency = computeInstrLatency(DefMI);
+  const unsigned DefaultDefLatency = TII->defaultDefLatency(SchedModel, *DefMI);
+
   if (!hasInstrSchedModel() && !hasInstrItineraries())
-    return TII->defaultDefLatency(SchedModel, *DefMI);
+    return InstrLatency;
```

Comment:
Is the change here intentional?

The commit message gives the impression that this commit mostly is a refactoring. But as far as I can tell we used to return `DefaultDefLatency `here rather than `InstrLatency`.

Reply:
This is handled by `computeInstrLatency`.

Reply:
But `computeInstrLatency` is for example always making an early out for `MI->isBundle()`. Are you sure that `InstrLatency==DefaultDefLatency` here?

Reply:
Yeah, I'm not sure. I'll fix this shortly. Thanks!